homeassistant/components/hive.py

```diff
@@ -0,0 +1,1000 @@
+"""Hive Integration - Platform."""
+import logging
+from datetime import datetime
+from homeassistant.helpers.discovery import load_platform
+
+REQUIREMENTS = ['pyhiveapi==0.0.41']
+
+HGO = None
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = 'hive'
+
+
+class HivePlatformData:
+    """Initiate Hive PlatformData Class."""
+
+    min_max_data = {}
+
+
+class HiveSession:
+    """Initiate Hive Session Class."""
+
+    username = ""
+    password = ""
+    platform_data = HivePlatformData()
+    logging = False
+    hass = None
+    core = None
+    heating = None
+    hotwater = None
+    light = None
+    sensor = None
+    switch = None
+
+
+HSC = HiveSession()
+
+
+def fire_bus_event(node_id, device_type):
+    """Fire off an event if some data has changed."""
+    fire_events = True
+    if fire_events:
+        HSC.hass.bus.fire('Event_Hive_NewNodeData', {device_type: node_id})
+
+
+def hive_api_get_nodes_rl(node_id, device_type):
+    """Get latest data for Hive nodes - rate limiting."""
+    nodes_updated = HSC.core.hive_api_get_nodes_rl(node_id)

```

Comment:
These wrapper functions should become obsolete, since the entities will be able to access the session from hass.data.

Reply:
wrapper functions removed.